src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -661,17 +662,29 @@ public void addEnvVars(List<TemplateEnvVar> envVars) {
         }
     }
 
+    @SuppressWarnings("unused") // Used by jelly
+    @Restricted(DoNotUse.class) // Used by jelly
+    public boolean hasManagePermission() {
+        StaplerRequest2 request = Stapler.getCurrentRequest2();
+        if (request != null) {
+            PodTemplateGroup groupFromRequest = request.findAncestorObject(PodTemplateGroup.class);
+            if(groupFromRequest!=null) {
+                return groupFromRequest.hasManagePermission();
+            };
+        }
+        return Jenkins.get().hasPermission(Jenkins.MANAGE);
+    }
+
     /**
      * Deletes the template.
      */
     @POST
     public HttpResponse doDoDelete(@AncestorInPath PodTemplateGroup owner) throws IOException {
-        Jenkins j = Jenkins.get();
-        j.checkPermission(Jenkins.MANAGE);
         if (owner == null) {
             throw new IllegalStateException("Cloud could not be found");
         }
         owner.removeTemplate(this);
+        Jenkins j = Jenkins.get();
         j.save();
```

Comment:
Seems a pre-existing bug: the owner should extend `Saveable` as it may not be persisted part of Jenkins root object.

Reply:
PodTemplate does `implements` it 

```
public class PodTemplate extends AbstractDescribableImpl<PodTemplate> implements Serializable, Saveable {
```

Reply:
`PodTemplate` is stored in a `PodTemplateGroup`. When this group a `Cloud` it is persisted in Jenkins so it makes no difference, but we have use cases where it is actually within an `Item`.

Reply:
I am sorry I am not sure what that means in terms of consequences or what I need to change to fix/enhance the situation.